### DISCUSSION TITLE
Implementing Programmable Filter ParaView in PyVista

### DISCUSSION THREAD

**Discussion by paolopiers** [pyvista/pyvista#5866] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

I am trying to port some Fenics Legacy 2019 code (from now on *old code*) into Fenicsx.
One of the operations the old code performs is to save the mesh in `.pvd` format. This is a minimal working example:

```python
from ast import Expression
from pyclbr import Function
from fenics import *
from math import *
from mshr import *
from dolfin import *
#from scipy.integrate import dblquad  
#from scipy.interpolate import griddata
#from ufl import Form, TestFunction, TrialFunction, inner, dx 
import sympy as sym
import matplotlib.pyplot as plt
import numpy as np
import copy
import os
p1 = Point(0,0)
p2 = Point(np.pi,1)
nx = 64
ny = 64
mesh = RectangleMesh(p1,p2,nx,ny)
file = File('MeshTest.pvd')
file << mesh
```

After saving the mesh file in `.pvd` format, I import it onto ParaView and I apply the following programmable filter to convert the rectangula domain into a cylinder. 

This kind of approach is way more amendable than the Cartesian approach for the mathematical analysis of models in elasticity.

```python
from paraview import vtk
import math
R=1.0
H=math.pi
pdi = self.GetUnstructuredGridInput()
pdo = self.GetOutput()
newPoints = vtk.vtkPoints()
numPoints = pdi.GetNumberOfPoints()
for i in range(0,numPoints):
    coord = pdi.GetPoint(i)
    x, y, z = coord[:3]
    s = R*cos(x)
    t = H*y
    z = -R*sin(x)+1.5
    newPoints.InsertPoint(i, s, t, z)
pdo.SetPoints(newPoints)
print("Done!")`
```

### Description

I would like to implement this programmable filter directly in PyVista, without having to resort to ParaView.

How would you recommend me to proceed?

Many thanks in advance.

### System Information

```shell
N/A
```

**Comment by MatthewFlamm** [comment, marked as answer]
Take a look at https://github.com/pyvista/pyvista/discussions/4914#discussioncomment-7037610 for an example of how to generate a sphere similar to this.
